### ISSUE TITLE
improve panicking

### ISSUE THREAD

**Issue by skanehira** [jesseduffield/lazygit#396]
**Is your feature request related to a problem? Please describe.**
When I try to edit or open a file, then will panicking.

```
2019/03/16 20:12:27 An error occurred! Please create an issue at https://github.com/jesseduffield/lazygit/issues

*errors.errorString No changed files
/Users/skanehira/dev/go/src/github.com/jesseduffield/lazygit/pkg/gui/gui.go:55 (0x18006f2)
        (*Gui).GenerateSentinelErrors: ErrNoFiles:    errors.New(gui.Tr.SLocalize("NoChangedFiles")),
/Users/skanehira/dev/go/src/github.com/jesseduffield/lazygit/pkg/gui/gui.go:184 (0x1800c14)
        NewGui: gui.GenerateSentinelErrors()
/Users/skanehira/dev/go/src/github.com/jesseduffield/lazygit/pkg/app/app.go:125 (0x182a440)
        NewApp: app.Gui, err = gui.NewGui(app.Log, app.GitCommand, app.OSCommand, app.Tr, config, app.Updater)
/Users/skanehira/dev/go/src/github.com/jesseduffield/lazygit/main.go:48 (0x182ae4d)
        main: app, err := app.NewApp(appConfig)
/usr/local/Cellar/go/1.12/libexec/src/runtime/proc.go:200 (0x102eecc)
        main: fn()
/usr/local/Cellar/go/1.12/libexec/src/runtime/asm_amd64.s:1337 (0x10599d1)
        goexit: BYTE    $0x90   // NOP
```

**Describe the solution you'd like**
Catch the error and solve it.

**Describe alternatives you've considered**
An error message window is displayed instead of panic.

**Additional context**
Nothing, I'll try improve this.

**Comment by skanehira** [comment]
There looks like just return a gui.Errors.ErrNoFiles.

```go
func (gui *Gui) handleEnterFile(g *gocui.Gui, v *gocui.View) error {
	file, err := gui.getSelectedFile(g)
	if err != nil {
		if err != gui.Errors.ErrNoFiles {
			return err
		}
		return nil
	}
	if file.HasInlineMergeConflicts {
		return gui.handleSwitchToMerge(g, v)
	}
	if !file.HasUnstagedChanges || file.HasMergeConflicts {
		return gui.createErrorPanel(g, gui.Tr.SLocalize("FileStagingRequirements"))
	}
	if err := gui.changeContext("main", "staging"); err != nil {
		return err
	}
	if err := gui.switchFocus(g, v, gui.getMainView()); err != nil {
		return err
	}
	return gui.refreshStagingPanel()
}
```

**Comment by jesseduffield** [comment]
Good pickup! Yep, looks like returning `gui.createErrorPanel(gui.g, err.Error())` is the right thing to do here

**Comment by skanehira** [comment]
It looks like fixed this commit d5f617e
Thank you :)
